docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+    - [1.3.1. Data Mode](#131-data-mode)
+      - [Supplemental: Error Handling in the SDK](#supplemental-error-handling-in-the-sdk)
+      - [Adding a custom method: `setup_data()`](#adding-a-custom-method-setup_data)
+    - [](#)
+
+
+## 1.1. Prerequisites
+
+While we do our best to provide enough information here for anyone with general scripting experience to follow along, the recommended prereqs are provided below for general debugging and troubleshooting knowledge outside of the scope possible here:
+
+* Tensorflow
+  * Keras
+  * Datasets
+* React and Typescript
+* General understanding or knowledge of Neural Networks and AI/ML algorithms.
+
+## 1.2. Overview
+
+The tool we create will allow a user to generate a Text Classifier NN(Neural Network) from scratch, train it using User-defined datasets, evaluate training, and then predict with a "production" version of said model.
+As mentioned, we'll aim to recreate Tensorflow's Official tutorial model as a workflow tool. If you run into TensorFlow related issues, we recommend you review and cross-check the code provided in the tutorial.
+
+With that in mind, we create a "naive" version of our text classifier tool.
+That is, a tool without any UI to ensure critical functions are running as intended.
+To achieve this, we also mock data and relevant calls to represent what will come from the UI SDK to the Python backend and what the backend will send to the UI SDK via `save_config()` (link to docs here). 
+Next, we build our front end using the UI SDK using its provided `dev-harness`.
+Once the back and front ends are complete, we then bundle and deploy using the CLI to do end-to-end testing.
+Finally, we package our dependencies and source for deployment using `create-yxi`.
+
+The tool will have 4 "modes": `DATA`, `PREVIEW`, `TRAIN`, and `PREDICT`. 
+
+### 1.2.1. Data
+
+In `DATA` mode, our tool will pull down and prep whatever data the User provides via input fields we'll define.
+It will then save the data as a TensorFlow dataset, ready for the next mode.
+
+### 1.2.2. Preview
+
+In `PREVIEW` mode, our tool will then use the data provided and any available model information to show a "snapshot" of our data and how we will feed it to the model.
+This snapshot will include dynamically populated text classifier information like token translations for our classifier when available.
+
+### 1.2.3. TRAIN
+
+This mode will take the prepped data and call the model's training function.
+Additionally, the User may use the supplied fields to adjust model parameters, train using an existing model, or create an entirely new one.
+Finally, the tool will return a training results report to the UI and export a "production" version of the model.
+
+### 1.2.4. Predict
+
+In `PREDICT` mode, the tool will take text input via the input anchor, then run predictions on them and output those results to its output anchor.
```

Comment:
TW review:

In `PREDICT` mode, the tool takes text input via the input anchor, then runs predictions on the input and outputs the results to its output anchor.